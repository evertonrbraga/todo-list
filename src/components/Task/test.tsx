import { render, screen } from 'project-testing-library'
import { Task } from '.'

describe('<Task />', () => {
  it('should snapshot the Task component', () => {
    const { container } = render(<Task />)
    expect(container.firstChild).toMatchSnapshot()
  })

  it('should check if the empty layout is rendering correctly', () => {
    render(<Task />)
    const svgImg = 'test-file-stub'
    const title = screen.getByText('Você ainda não tem tarefas cadastradas')
    const text = screen.getByText('Crie tarefas e organize seus itens a fazer')
    const clipboardIcon = screen.getByAltText('Ícone prancheta')
    expect(title).toBeInTheDocument()
    expect(text).toBeInTheDocument()
    expect(clipboardIcon).toHaveAttribute('src', svgImg)
  })
})
