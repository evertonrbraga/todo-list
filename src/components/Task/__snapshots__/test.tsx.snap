// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Task /> should snapshot the Task component 1`] = `
.c0 {
  background: #262626;
}

<main
  class="c0"
>
  <h1>
    Você ainda não tem tarefas cadastradas
  </h1>
</main>
`;
